crates/fuel-core/src/service/genesis/importer/import_task.rs

```diff
@@ -553,32 +632,104 @@ mod tests {
         }
     }
 
-    impl TransactableStorage<BlockHeight> for BrokenTransactions {
+    impl<DbDesc> TransactableStorage<DbDesc::Height> for BrokenTransactions<DbDesc>
+    where
+        DbDesc: DatabaseDescription,
+    {
         fn commit_changes(
             &self,
-            _: Option<BlockHeight>,
+            _: Option<DbDesc::Height>,
             _: Changes,
         ) -> StorageResult<()> {
             Err(anyhow::anyhow!("I refuse to work!").into())
         }
     }
 
-    #[test]
-    fn tx_commit_failure_is_propagated() {
+    #[test_case::test_case(
+        GenesisDatabase::new(Arc::new(BrokenTransactions::<OnChain>::default())),
+        GenesisDatabase::default()
+    ; "broken on chain db")]
+    #[test_case::test_case(
+        GenesisDatabase::default(),
+        GenesisDatabase::new(Arc::new(BrokenTransactions::<OffChain>::default()))
+    ; "broken off chain db")]
+    fn on_chain_tx_commit_failure_is_propagated(
+        on_chain: GenesisDatabase<OnChain>,
+        off_chain: GenesisDatabase<OffChain>,
+    ) {
         // given
         let groups = TestData::new(1);
-        let runner = ImportTask::new(
-            TestHandler::new(|_, _| Ok(())),
-            groups.as_ok_groups(),
-            GenesisDatabase::new(Arc::new(BrokenTransactions::new())),
+
+        // TODO: check off chain as well
+        // when
+        let result = import_entries(
+            never_cancel(),
+            Spy::default().default_importer(),
+            groups.as_ok_groups(0),
+            on_chain,
+            off_chain,
             ProgressReporter::default(),
         );
 
+        // then
+        assert!(result.is_err());
+    }
+
+    #[test_case::test_case(None, Some(0) ; "on chain reverted at start")]
+    #[test_case::test_case(Some(0), None; "off chain reverted at start")]
+    #[test_case::test_case(Some(0), Some(1); "on chain reverted")]
+    #[test_case::test_case(Some(1), Some(0); "off chain reverted")]
+    #[test_case::test_case(Some(3), Some(1); "off chain reverted multiple times")]
+    #[test_case::test_case(Some(1), Some(3); "on chain reverted multiple times")]
+    fn can_recover_when_both_tx_dont_succeed_together(
```

Comment:
I think this test could have a clearer name.

What "can_recover"? What are the "tx"s here? I don't see anything called "tx" in this test. I feel like the name should have more to do with reconciling the groups being processed.

Reply:
> What "can_recover"?

Importing/genesis/regenesis after the process exists due to an error.

> What are the "tx"s here?

Database transactions. Two of them, one for on-chain and another on for off chain databases. Only when successfully committed the import progress is going to be incremented. They will be committed only if nothing failed during the import.

> I feel like the name should have more to do with reconciling the groups being processed.

Maybe `synchronizes_imports_upon_previous_failure` ? To say that even if a batch succeeds on one of the databases and fails on another, the next time the import is run the imports are first going to be synchronized. 

Thing is I don't want to commit to `synchronizes` because that is not a requirement. If somebody parallelized this they can very well have both databases continue importing even though they're not handling the same batch at the same time.

I guess failed batches will be retried after failure. Something along those lines?